File: old-projects/vector-defence-silverlight-port/src/Monsters/BallMonster.ts
import { Point } from "../Point";
import { Monster } from "./Monster";

export class BallMonster extends Monster {
  private static readonly diameter = 15;

  constructor(x: number, y: number, path: Point[]) {
    super(x, y, path, 1.5);
    this.bounty = 20;
    this.color = "#00ffff";
    this.radius = BallMonster.diameter / 2;
    this.life = this.originalLife = 200;
  }

  override render(context: CanvasRenderingContext2D): void {
    context.save();
    context.globalAlpha = 1;
    context.fillStyle = this.damageAlpha > 0 ? `rgba(128,0,128,${this.damageAlpha})` : "#000";
    context.strokeStyle = this.color;
    context.lineWidth = 1.5;
    context.beginPath();
    context.arc(this.x, this.y, this.radius, 0, Math.PI * 2);
    context.fill();
    context.stroke();
    context.restore();
    super.render(context);
  }
}

File: old-projects/vector-defence-silverlight-port/original-csharp-source/Monsters/BallMonster.cs
using System.Collections.Generic;
using System.Windows;
using System.Windows.Media;
using System.Windows.Shapes;

namespace VectorDefenceSL.Monsters
{
    public class BallMonster : Monster
    {
        private const int DIAMETER = 15;

        public BallMonster(double x, double y, IList<Point> path) : base(x, y, path, 1.5)
        {
            Bounty = 20;
            Color = Colors.Cyan;
            Radius = DIAMETER / 2;
            life = originalLife = 200;
            mainElement = new Ellipse
            {
                StrokeThickness = 1.5,
                Stroke = new SolidColorBrush(Color),
                Fill = new SolidColorBrush(Colors.Black),
                Width = DIAMETER,
                Height = DIAMETER,
            };

            Game.AddElement(mainElement);
        }
        
        public override void Render()
        {
            Move(X - Radius, Y - Radius);
            base.Render();
        }
    }
}
